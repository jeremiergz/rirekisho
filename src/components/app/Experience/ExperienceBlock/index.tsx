import React, { FunctionComponent } from 'react';
import Box from '../../../primitives/Box';
import FlexBox from '../../../primitives/FlexBox';
import Grid, { GridProps } from '../../../primitives/Grid';
import Text from '../../../primitives/Text';
import ProjectBlock from './ProjectBlock';

const months = {
  0: 'JAN',
  1: 'FEB',
  2: 'MAR',
  3: 'APR',
  4: 'MAY',
  5: 'JUN',
  6: 'JUL',
  7: 'AUG',
  8: 'SEP',
  9: 'OCT',
  10: 'NOV',
  11: 'DEC',
};

function getFormattedDate(dateString: string) {
  const date = dateString && new Date(dateString);
  return dateString ? `${months[date.getMonth()]} ${date.getFullYear()}` : 'PRESENT';
}

const ExperienceBlock: FunctionComponent<ExperienceBlockProps> = ({ item, ...rest }) => {
  const endDate = getFormattedDate(item.endDate);
  const startDate = getFormattedDate(item.startDate);
  return (
    <Grid
      alignItems="stretch"
      flexDirection="column"
      justifyContent="flex-start"
      marginBottom={4}
      variant="item"
      {...rest}
    >
      <FlexBox alignItems="flex-start" justifyContent="space-between" marginBottom={2}>
        <Box color="primary" fontSize={18} fontWeight="bold" marginRight={3}>
          <Text display={{ _: 'block', mobileM: 'inline' }} whiteSpace="nowrap">
            {startDate}
            {' - '}
          </Text>
          <Text display={{ _: 'block', mobileM: 'inline' }} whiteSpace="nowrap">
            {endDate}
          </Text>
        </Box>
        <Box color="secondary" fontWeight="bold" textAlign="right">
          <Text fontSize={18}>{item.company}</Text> |{' '}
          <Text fontSize={16} fontStyle="italic">
            {item.companySector}
          </Text>
        </Box>
      </FlexBox>
      {item.projects.map((project, index) => (
        <ProjectBlock
          item={project}
          key={project.description}
          marginBottom={index < item.projects.length - 1 ? '24px' : 0}
        />
      ))}
    </Grid>
  );
};

ExperienceBlock.displayName = 'ExperienceBlock';

export type ExperienceBlockProps = Omit<GridProps, 'variant'> & {
  item: Models.Experience;
};
export default ExperienceBlock;
