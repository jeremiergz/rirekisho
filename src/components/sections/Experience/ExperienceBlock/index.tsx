import Box from 'components/primitives/Box';
import FlexBox from 'components/primitives/FlexBox';
import Grid, { GridProps } from 'components/primitives/Grid';
import Text from 'components/primitives/Text';
import { useTheme } from 'components/providers/ThemeProvider';
import Img, { FixedObject } from 'gatsby-image';
import React from 'react';
import ProjectBlock from './ProjectBlock';

const months = {
  0: 'JAN',
  1: 'FEB',
  2: 'MAR',
  3: 'APR',
  4: 'MAY',
  5: 'JUN',
  6: 'JUL',
  7: 'AUG',
  8: 'SEP',
  9: 'OCT',
  10: 'NOV',
  11: 'DEC',
};

function getFormattedDate(dateString: string) {
  const date = dateString && new Date(dateString);
  return dateString ? `${months[date.getMonth()]} ${date.getFullYear()}` : 'PRESENT';
}

const ExperienceBlock: React.FC<ExperienceBlockProps> = ({ companyImg, item, ...rest }) => {
  const { theme } = useTheme();
  const endDate = getFormattedDate(item.endDate);
  const startDate = getFormattedDate(item.startDate);
  return (
    <Grid
      alignItems="stretch"
      flexDirection="column"
      justifyContent="flex-start"
      marginBottom={4}
      variant="item"
      {...rest}
    >
      <FlexBox
        flexDirection={{ _: 'column', tablet: 'row' }}
        alignItems={{ _: 'flex-start', tablet: 'center' }}
        justifyContent="space-between"
        marginBottom={2}
      >
        <Box color="primary" fontSize={18} fontWeight="bold" marginRight={3}>
          <Text whiteSpace="nowrap">
            {startDate}
            {' - '}
          </Text>
          <Text whiteSpace="nowrap">{endDate}</Text>
        </Box>
        <FlexBox
          alignItems="center"
          color="secondary"
          flexDirection={{ _: 'row', tablet: 'row-reverse' }}
          fontWeight="bold"
          marginY={{ _: 2, tablet: 0 }}
          textAlign={{ _: 'left', tablet: 'right' }}
        >
          <FlexBox
            backgroundColor={theme.type === 'dark' ? 'text' : 'none'}
            borderRadius={16}
            paddingX={2}
            paddingY={1}
          >
            {companyImg ? (
              <Img fixed={companyImg} />
            ) : (
              <Text color="dark" fontSize={18}>
                {item.company}
              </Text>
            )}
          </FlexBox>
          <Text marginX={2}>|</Text>
          <FlexBox flex={1}>
            <Text fontSize={16} fontStyle="italic">
              {item.companySector}
            </Text>
          </FlexBox>
        </FlexBox>
      </FlexBox>
      {item.projects.map((project, index) => (
        <ProjectBlock
          item={project}
          key={project.description}
          marginBottom={index < item.projects.length - 1 ? '24px' : 0}
        />
      ))}
    </Grid>
  );
};

ExperienceBlock.displayName = 'ExperienceBlock';

export type ExperienceBlockProps = Omit<GridProps, 'variant'> & {
  companyImg?: FixedObject;
  item: Models.Experience;
};
export default ExperienceBlock;
