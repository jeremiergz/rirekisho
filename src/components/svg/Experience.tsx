import React, { ComponentProps } from 'react';

const Experience: React.FC<ExperienceProps> = props => {
  return (
    <svg fill="#fff" height="24" width="24" {...props}>
      <path d="M0 0h24v24H0z" fill="none"></path>
      <path d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"></path>
    </svg>
  );
};

Experience.displayName = 'Experience';

export type ExperienceProps = ComponentProps<'svg'>;
export default Experience;
