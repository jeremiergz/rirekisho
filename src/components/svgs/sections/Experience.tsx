import SVG, { SVGProps } from 'components/SVG';
import React from 'react';

const Experience: React.FC<ExperienceProps> = props => {
  return (
    <SVG {...props}>
      <path d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z" />
    </SVG>
  );
};

Experience.displayName = 'Experience';

export type ExperienceProps = SVGProps;
export default Experience;
