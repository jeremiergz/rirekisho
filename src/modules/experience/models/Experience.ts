import { IGatsbyImageData } from 'gatsby-plugin-image';
import Project from './Project';

type ExperienceProps = {
  company: {
    img: {
      src: IGatsbyImageData;
    } | null;
    name: string | null;
    sector: string;
    website: string | null;
  };
  projects: Project[];
  timeline: {
    endDate: string;
    startDate: string;
  };
};

interface Experience extends ExperienceProps {}

class Experience {
  constructor({ company, projects, timeline }: ExperienceProps) {
    this.company = company;
    this.projects = projects;
    this.timeline = timeline;
  }
}

export default Experience;
