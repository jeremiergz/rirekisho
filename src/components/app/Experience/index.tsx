import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import Layout from '../../common/Layout';
import ExperienceBlock from './ExperienceBlock';
import Grid from '../../primitives/Grid';

const Experience = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    experienceNodes: { nodes: experiences },
  } = useStaticQuery<GraphQL.ExperienceData>(graphql`
    query ExperienceData {
      experienceNodes: allExperiencesJson {
        nodes {
          company
          companySector
          endDate
          img
          projects {
            client
            clientSector
            description
            name
            tasks
            technologies
          }
          startDate
        }
      }
    }
  `);
  return (
    <Layout.Section ref={ref} title="experience">
      <Layout.Content columnDirection="column-reverse">
        <Grid gridColumnGap={{ _: 32, tablet: 64 }} gridTemplateColumns="auto auto" variant="container">
          {experiences
            .sort((a, b) => b.startDate.localeCompare(a.startDate))
            .map(item => (
              <ExperienceBlock key={item.startDate} gridColumn={{ _: 'span 2', laptopS: 'span 1' }} item={item} />
            ))}
        </Grid>
      </Layout.Content>
    </Layout.Section>
  );
});

Experience.displayName = 'Experience';

export default Experience;
