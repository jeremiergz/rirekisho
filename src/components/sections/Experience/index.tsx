import Layout from 'components/Layout';
import Grid from 'components/primitives/Grid';
import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import ExperienceBlock from './ExperienceBlock';

const Experience = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    companyImgNodes: { nodes: companyImgs },
    experienceNodes: { nodes: experiences },
  } = useStaticQuery<GraphQL.ExperienceDataQuery>(graphql`
    query ExperienceData {
      companyImgNodes: allImageSharp(filter: { original: { src: { regex: "/company/" } } }) {
        nodes {
          fixed(height: 24, quality: 100) {
            originalName
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
      experienceNodes: allExperiencesJson {
        nodes {
          company
          companySector
          endDate
          img
          projects {
            client
            clientSector
            description
            name
            tasks
            technologies
          }
          startDate
        }
      }
    }
  `);
  const companyImgsIndex = companyImgs.reduce((acc, img) => {
    acc[img.fixed.originalName] = img.fixed;
    return acc;
  }, {});
  return (
    <Layout.Section ref={ref} title="experience">
      <Layout.Content columnDirection="column-reverse">
        <Grid gridColumnGap={{ _: 32, tablet: 64 }} gridTemplateColumns="auto auto" variant="container">
          {experiences
            .sort((a, b) => b.startDate.localeCompare(a.startDate))
            .map(item => {
              const img = companyImgsIndex[item.img];
              return (
                <ExperienceBlock
                  companyImg={img}
                  key={item.startDate}
                  gridColumn={{ _: 'span 2', laptopS: 'span 1' }}
                  item={item as Models.Experience}
                />
              );
            })}
        </Grid>
      </Layout.Content>
    </Layout.Section>
  );
});

Experience.displayName = 'Experience';

export default Experience;
