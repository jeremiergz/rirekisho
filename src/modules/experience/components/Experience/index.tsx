import Content from '@/common/components/layout/Content';
import Section from '@/common/components/layout/Section';
import clsx from 'clsx';
import React, { forwardRef } from 'react';
import useExperiencesData from '../../hooks/useExperiencesData';
import ExperienceBlock from './ExperienceBlock';

const Experience = forwardRef<HTMLDivElement>(function Experience(_, ref): JSX.Element {
  const experiences = useExperiencesData();

  return (
    <Section className="my-0" ref={ref} title="experience">
      <Content reverse>
        <div>
          {experiences
            .sort((a, b) => new Date(b.timeline.startDate).getTime() - new Date(a.timeline.startDate).getTime())
            .map((item, index) => (
              <ExperienceBlock
                className={clsx(index !== experiences.length - 1 && 'mb-8 print:mb-6')}
                item={item}
                key={item.timeline.startDate}
              />
            ))}
        </div>
      </Content>
    </Section>
  );
});

export default Experience;
