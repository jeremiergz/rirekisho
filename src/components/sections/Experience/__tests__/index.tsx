import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import Experience from '..';
import mockData from './mock.json';

describe('components/sections/Experience Test Suite', () => {
  beforeAll(() => {
    const graphql = jest.spyOn(Gatsby, 'graphql');
    graphql.mockImplementation(() => '');
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => mockData);
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={theme}>
        <Experience />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
