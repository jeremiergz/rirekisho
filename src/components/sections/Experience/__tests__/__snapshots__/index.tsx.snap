// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sections/Experience Test Suite matches snapshot 1`] = `
<section
  className="sc-AxjAm sc-AxirZ jFiolx"
>
  <div
    className="sc-AxjAm eiESPf"
    fontSize={28}
    fontWeight="bold"
  >
    <span
      className="sc-AxiKw ggvGYh"
      color="secondary"
    >
      exp
    </span>
    <span
      className="sc-AxiKw jncuiT"
      color="primary"
    >
      erience
    </span>
  </div>
  <div
    className="sc-AxjAm fhEmi"
    display="flex"
    width="100%"
  >
    <div
      className="sc-AxjAm sc-AxirZ cusFct"
    >
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              OCT 2018
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              PRESENT
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              Sigfox
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              Wireless Networks
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm gFKxpe"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                XDoG
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Kubernetization & move to GCP of Web platforms
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                Docker | JavaScript | Kubernetes | GCP | Istio
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Migrate web platforms to Kubernetes on Google Cloud Platform
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Adapt APIs for K8s: health probes, Prometheus metrics, etc
          </span>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Features development & maintenance of Web platforms
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                Node.js | React | JavaScript | Styled-Components | Gatsby.js
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Back-end & front-end engineering & development
          </span>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              JUN 2018
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              SEP 2018
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              Capgemini
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              IT Services
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Client
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                Airbus Defence & Space
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Space Systems
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                NemEO - DiginEO
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Web framework - Web portal
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                React.js | Redux | Node.js | TypeScript | Docker
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Technical manager of a 3 developer team
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Front-end development: React.js & Redux
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Maintenance of the Architectural Design Document
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Deployment industrialization using Docker
          </span>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              JUN 2017
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              JUN 2018
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              CS
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              Aerospace Industry
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Client
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                Airbus Defence & Space
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Space Systems
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                Pulsar
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Ground-Segment simulators, validation & testing web interface
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                Node.js | TypeScript | Angular | Docker | PostgreSQL | Java 8 | Spring Boot
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Use case supervisor on Postgres DB engineering & implementation
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Development of multiple micro-services & components: - back-end using Node.js & TypeScript - front-end using Angular v6
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Micro-services dockerization
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            CLI utilities development
          </span>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              JAN 2017
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              MAY 2017
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              CS
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              Aerospace Industry
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Client
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                CNES
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                French Space Agency
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                Trafic
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Customizable transfer tool using native File System, FTP & SFTP
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                Java 6 | JsCh | Jenkins
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Development of a new transfer protocol using native File System
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Maintenance & quality controls: refactoring of several classes
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Jenkins CI/CD project integration
          </span>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              NOV 2016
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              DEC 2016
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              SQLI
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              IT Services
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Client
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                Omnium Finance
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Financial Services
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Client base management Web interface for tablets and PCs
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                Angular.js | JavaScript | Docker | Spring Boot
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Back-end development using Java 8 with Spring Boot & JPA
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Front-end development using Angular.js, Bootstrap & SASS
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Docker initiation & Spring Boot container creation
          </span>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-AxirZ jnaqMB"
      >
        <div
          className="sc-AxjAm rtJDQ"
          display="flex"
        >
          <div
            className="sc-AxjAm lpbfZv"
            color="primary"
            fontSize={18}
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              JUL 2016
               - 
            </span>
            <span
              className="sc-AxiKw fYurIJ"
              display={
                Object {
                  "_": "block",
                  "mobileM": "inline",
                }
              }
            >
              OCT 2016
            </span>
          </div>
          <div
            className="sc-AxjAm gGvlbv"
            color="secondary"
            fontWeight="bold"
          >
            <span
              className="sc-AxiKw gTWeri"
              fontSize={18}
            >
              PragmaNTIC
            </span>
             |
             
            <span
              className="sc-AxiKw kEolGq"
              fontSize={16}
              fontStyle="italic"
            >
              Startup
            </span>
          </div>
        </div>
        <div
          className="sc-AxjAm diSKPk"
        >
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Client
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw cBoXSN"
              >
                RoboCARE Lab
                 | 
              </span>
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Home Robotics Services
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Project
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw bZBxFE"
                color="normal"
                fontStyle="italic"
              >
                Client base management Web interface for tablets and PCs
              </span>
            </div>
          </div>
          <div
            className="sc-AxjAm exrHeN"
            display="flex"
          >
            <span
              className="sc-AxiKw hOsDPP"
              fontWeight="bold"
            >
              Techs
              :
            </span>
            <div
              className="sc-AxjAm fwkkgR"
              fontWeight="bolder"
            >
              <span
                className="sc-AxiKw jVbjRr"
                color="secondary"
                fontStyle="italic"
              >
                PHP | Symfony 3 | JavaScript | MySQL
              </span>
            </div>
          </div>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Engineering of a generic MySQL DB for 3 different web portals
          </span>
          <span
            className="sc-AxiKw kzTKyK"
            color="#575757"
            display="block"
            fontWeight="bolder"
          >
            • 
            Web portal development in PHP using the Symfony 3 framework
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
